libdnf/base/base.cpp

```diff
@@ -190,6 +190,12 @@ void Base::setup() {
 
     get_vars()->load(vars_installroot, config.get_varsdir_option().get_value());
 
+    if (!config.get_forcearch_option().get_value().empty()) {
+        config.get_ignorearch_option().set(Option::Priority::RUNTIME, true);
+        get_vars()->set("arch", config.get_forcearch_option().get_value(), libdnf::Vars::Priority::CONFIG);
+        get_vars()->set("basearch", config.get_forcearch_option().get_value(), libdnf::Vars::Priority::CONFIG);
```

Comment:
I think we should add another logic here  or define what is allowed for forcearch.
My system can detect that I have `amd64`, then arch == `amd64`. And base arch is generated from that value  using internal get_base_arch(). Base arch is important for repository path substitution therefore using any value will lead into repository paths that are not available.

Dnf4 validates value and convert it from arch to basearch:
```
 sudo dnf --forcearch amd64 install acpi
Fedora 38 - x86_64 - Updates                                                                                                                                                                            24 kB/s |  19 kB     00:00    
Fedora 38 - x86_64 - Updates                                                                                                                                                                           2.5 MB/s | 2.5 MB     00:01    
Fedora Modular 38 - x86_64 - Updates                                                                                                                                                                    15 kB/s |  18 kB     00:01    
Dependencies resolved.
=======================================================================================================================================================================================================================================
 Package                                              Architecture                                           Version                                                      Repository                                              Size
=======================================================================================================================================================================================================================================
Installing:
 acpi                                                 x86_64                                                 1.7-20.fc38                                                  fedora                                                  28 k

Transaction Summary
=======================================================================================================================================================================================================================================
Install  1 Package

Total download size: 28 k
Installed size: 52 k
Is this ok [y/N]: 
```
```
sudo dnf --forcearch and64 install acpi
usage: dnf [-c [config file]] [-q] [-v] [--version] [--installroot [path]] [--nodocs] [--noplugins] [--enableplugin [plugin]] [--disableplugin [plugin]] [--releasever RELEASEVER] [--setopt SETOPTS] [--skip-broken] [-h]
           [--allowerasing] [-b | --nobest] [-C] [-R [minutes]] [-d [debug level]] [--debugsolver] [--showduplicates] [-e ERRORLEVEL] [--obsoletes] [--rpmverbosity [debug level name]] [-y] [--assumeno] [--enablerepo [repo]]
           [--disablerepo [repo] | --repo [repo]] [--enable | --disable] [-x [package]] [--disableexcludes [repo]] [--repofrompath [repo,path]] [--noautoremove] [--nogpgcheck] [--color COLOR] [--refresh] [-4] [-6]
           [--destdir DESTDIR] [--downloadonly] [--comment COMMENT] [--bugfix] [--enhancement] [--newpackage] [--security] [--advisory ADVISORY] [--bz BUGZILLA] [--cve CVES] [--sec-severity {Critical,Important,Moderate,Low}]
           [--forcearch ARCH]
Command line error: argument --forcearch: invalid choice: 'and64' (choose from 'aarch64', 'alpha', 'alphaev4', 'alphaev45', 'alphaev5', 'alphaev56', 'alphaev6', 'alphaev67', 'alphaev68', 'alphaev7', 'alphapca56', 'amd64', 'armv5tejl', 'armv5tel', 'armv5tl', 'armv6hl', 'armv6l', 'armv7hl', 'armv7hnl', 'armv7l', 'armv8hl', 'armv8l', 'athlon', 'geode', 'i386', 'i486', 'i586', 'i686', 'ia32e', 'ia64', 'loongarch64', 'mips', 'mips64', 'mips64el', 'mipsel', 'noarch', 'ppc', 'ppc64', 'ppc64iseries', 'ppc64le', 'ppc64p7', 'ppc64pseries', 'riscv128', 'riscv32', 'riscv64', 's390', 's390x', 'sh3', 'sh4', 'sh4a', 'sparc', 'sparc64', 'sparc64v', 'sparcv8', 'sparcv9', 'sparcv9v', 'x86_64')

```